setup.py

```diff
@@ -67,6 +67,7 @@ def generate_proto(source):
         'numpy>=1.20',
         'opt_einsum',
         'scipy>=1.5',
+        'typing_extensions>=4.5.0',
```

Comment:
What's the reason for adding this as a compulsory dependency?

Reply:
I could make it a dev-dependency.  Would that be better?

Reply:
I made it a dev-dependency in a subsequence change (within this PR) so that you can assess whether you like that better.

Reply:
I think it's fine to add as a dependency, but I'm also curious why it's necessary

Reply:
@jakevdp We need `ParamSpec`, which was [added](https://peps.python.org/pep-0612/) in Python 3.10 (Jax's minimum version per NEP 29 is 3.8.  We could make it a dev-dependency (as I've now done) although I think it complicates the code.